src/Memory/MemoryInfoProducer.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "Memory/MemoryInfoProducer.h"
+
+#include <chrono>
+#include <thread>
+
+#include "MemoryUtils.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/Profiling.h"
+#include "capture.pb.h"
+
+namespace orbit_memory {
+
+using orbit_grpc_protos::TotalMemoryUsageInfo;
+
+void MemoryInfoProducer::Start() {
+  *exit_requested_ = false;
+  thread_ = std::make_shared<std::thread>(&MemoryInfoProducer::Run, this);
+}
+
+void MemoryInfoProducer::Stop() {
+  *exit_requested_ = true;
+  if (thread_ != nullptr && thread_->joinable()) {
+    thread_->join();
+  }
+  thread_.reset();
+}
+
+void MemoryInfoProducer::Run() {
+  pthread_setname_np(pthread_self(), "MemoryInfoProducer::Run");
```

Comment:
The thread name will not be set as it exceeds 15 characters.

Reply:
also - it is better to use the method from OrbitBase/ThreadUtils.h

Reply:
I changed to use methods from the ThreadUtils.h and set the thread name as `orbit_base::SetCurrentThreadName("MIP::Run")`.

This change is included in the commit *[fixup] Set thread name with the ThreadUtils method*.

Sorry again for the inconvenience brought by my rebasing :-(